sdks/python/apache_beam/io/tfrecordio.py

```diff
@@ -0,0 +1,274 @@
+      while True:
+        if not offset_range_tracker.try_claim(current_offset):
+          raise RuntimeError('Unable to claim position: %s' % current_offset)
+        record = _TFRecordUtil.read_record(file_handle)
+        if record is None:
+          return  # Reached EOF
+        else:
+          current_offset += _TFRecordUtil.encoded_num_bytes(record)
+          yield self._coder.decode(record)
+
+
+class ReadFromTFRecord(PTransform):
+  """Transform for reading TFRecord sources."""
+
+  def __init__(self,
+               file_pattern,
+               coder=coders.BytesCoder(),
+               compression_type=fileio.CompressionTypes.AUTO,
+               **kwargs):
+    """Initialize a ReadFromTFRecord transform.
+
+    Args:
+      file_pattern: The file pattern to read from as a local file path or a GCS
+        gs:// path. The path can contain glob characters (*, ?, and [...]
+        sets).
+      coder: Coder used to decode each record.
+      compression_type: Used to handle compressed input files. Typical value
+          is CompressionTypes.AUTO, in which case the file_path's extension will
+          be used to detect the compression.
+      **kwargs: optional args dictionary. These are passed through to parent
+        constructor.
+
+    Returns:
+      A WriteToTFRecord transform object.
+    """
+    super(ReadFromTFRecord, self).__init__(**kwargs)
+    self._args = (file_pattern, coder, compression_type)
+
+  def expand(self, pvalue):
+    return pvalue.pipeline | Read(_TFRecordSource(*self._args))
+
+
+class _TFRecordSink(fileio.FileSink):
+  """Sink for writing TFRecords files.
+
+  For detailed TFRecord format description see:
+    https://www.tensorflow.org/versions/master/api_docs/python/python_io.html#tfrecords-format-details
+  """
+
+  def __init__(self, file_path_prefix, coder, file_name_suffix, num_shards,
+               shard_name_template, compression_type):
+    """Initialize a TFRecordSink. See WriteToTFRecord for details."""
+
+    super(_TFRecordSink, self).__init__(
+        file_path_prefix=file_path_prefix,
+        coder=coder,
+        file_name_suffix=file_name_suffix,
+        num_shards=num_shards,
+        shard_name_template=shard_name_template,
+        mime_type='application/octet-stream',
+        compression_type=compression_type)
+
+  def write_encoded_record(self, file_handle, value):
+    _TFRecordUtil.write_record(file_handle, value)
+
+
+class WriteToTFRecord(PTransform):
+  """Transform for writing to TFRecord sinks."""
+
+  def __init__(self,
+               file_path_prefix,
+               coder=coders.BytesCoder(),
+               file_name_suffix='',
+               num_shards=0,
+               shard_name_template=fileio.DEFAULT_SHARD_NAME_TEMPLATE,
+               compression_type=fileio.CompressionTypes.AUTO,
+               **kwargs):
+    """Initialize WriteToTFRecord transform.
+
+    Args:
+
+      file_path_prefix: The file path to write to. The files written will begin
```

Comment:
I think the sentence that starts with `In most cases` is not necessary. We would not know what users will do with it. We can mention that this is a required argument unlike others.